csharp/examples/FlightAspServerExample/Services/InMemoryFlightServer.cs

```diff
@@ -0,0 +1,158 @@
+                schema,
+                descriptor,
+                new List<FlightEndpoint> { GetEndpoint(ticket, $"http://{context.Host}") },
+                numRows,
+                -1 // Unknown
+            ));
+            _flightData.tables.Add(ticket, newTable);
+
+            await responseStream.WriteAsync(new FlightPutResult("Table saved."));
+        }
+
+        public override async Task DoGet(
+            FlightTicket ticket,
+            FlightServerRecordBatchStreamWriter responseStream,
+            ServerCallContext context
+        )
+        {
+            if (!_flightData.tables.ContainsKey(ticket))
+            {
+                throw new RpcException(new Status(StatusCode.NotFound, "Flight not found."));
+            }
+            var table = _flightData.tables[ticket];
+
+            foreach (var batch in table) {
+                await responseStream.WriteAsync(batch);
+            }
+        }
+
+        public override async Task ListFlights(
+            FlightCriteria request,
+            IAsyncStreamWriter<FlightInfo> responseStream,
+            ServerCallContext context
+        )
+        {
+            foreach (var flight in _flightData.flights.Values)
+            {
+                await responseStream.WriteAsync(flight);
+            }
+        }
+
+        public override async Task<FlightInfo> GetFlightInfo(FlightDescriptor request, ServerCallContext context)
+        {
+            var key = DescriptorAsTicket(request);
+            if (_flightData.flights.ContainsKey(key))
+            {
+                return _flightData.flights[key];
+            }
+            else
+            {
+                throw new RpcException(new Status(StatusCode.NotFound, "Flight not found."));
+            }
+        }
+
+        public override async Task ListActions(
+            IAsyncStreamWriter<FlightActionType> responseStream, 
+            ServerCallContext context
+        )
+        {
+            await responseStream.WriteAsync(new FlightActionType("clear", "Clear the flights from the server"));
+        }
+
+        public override async Task DoAction(
+            FlightAction request, 
+            IAsyncStreamWriter<FlightResult> responseStream, 
+            ServerCallContext context
+        )
+        {
+            if (request.Type == "clear")
+            {
+                _flightData.flights.Clear();
+                _flightData.tables.Clear();
+            }
+            else
+            {
+                throw new RpcException(new Status(StatusCode.InvalidArgument, "Action does not exist."));
+            }
+        }
+
+        public override async Task<Schema> GetSchema(FlightDescriptor request, ServerCallContext context)
+        {
+            var info = await GetFlightInfo(request, context);
+            return info.Schema;
+        }
+
+        private FlightTicket DescriptorAsTicket(FlightDescriptor desc)
+        {
+            return new FlightTicket(desc.ToString());
```

Comment:
`FlightDescriptor` doesn't override ToString, so it just returns the Type name when you call `ToString()`. Is creating the same `FlightTicket`, no matter the instance, the correct thing here?

Reply:
Ah right. Good catch. Switching to `desc.Command`, which is the property we are passing.